File: custom_msg_pkg/src/custom_msg_sub_node.py
#!/usr/bin/env python3
#conding: utf-8


import rclpy
from custom_msgs.msg import CustomMsg

node = None

def custom_msg_callback(msg):
    global node
    node.get_logger().info("I am the custom msg sub node and received a message: ")
    node.get_logger().warn(f"grade: {msg.grade}")
    node.get_logger().warn(f"star: {msg.star}")
    node.get_logger().info(f"data: {msg.data}")


def main(args=None):
    global node
    rclpy.init(args=None)
    node = rclpy.create_node('custom_msg_sub_node')

    print("Hello World, I am the custom msg sub node! [Python]\n")

    sub_ultra_sound = node.create_subscription(CustomMsg, 'custom_msg_topic', custom_msg_callback, 10)

    rclpy.spin(node)

    node.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: custom_msg_pkg/src/custom_msg_sub_node.cpp
#include <rclcpp/rclcpp.hpp>
#include <custom_msgs/msg/custom_msg.hpp>

rclcpp::Node::SharedPtr node;

void custom_msg_callback(const std::shared_ptr<custom_msgs::msg::CustomMsg> msg)
{
    RCLCPP_INFO(node->get_logger(),
                "I am the custom msg sub node and received a message:");
    RCLCPP_WARN(node->get_logger(),
                "grade: %s",
                msg->grade.c_str());
    RCLCPP_WARN(node->get_logger(),
                "start: %d",
                msg->star);
    RCLCPP_INFO(node->get_logger(),
                "data: %s",
                msg->data.c_str());
}



int main(int argc, char const *argv[])
{
    rclcpp::init(argc, argv);

    printf("Hello World, I am the custom msg sub node!\n");

    node = rclcpp::Node::make_shared("custom_msg_sub_node");

    rclcpp::Subscription<custom_msgs::msg::CustomMsg>::SharedPtr sub = node->create_subscription<custom_msgs::msg::CustomMsg>(
        "custom_msg_topic",
        10,
        custom_msg_callback);


    rclcpp::spin(node);

    return 0;
}
